src/libstore/filetransfer.cc

```diff
@@ -846,7 +887,7 @@ ref<FileTransfer> getFileTransfer()
     static ref<curlFileTransfer> fileTransfer = makeCurlFileTransfer();
 
     if (fileTransfer->state_.lock()->quit)
-        fileTransfer = makeCurlFileTransfer();
+        fileTransfer->restart();
```

Comment:
```suggestion
        fileTransfer->restart(state);
```

Reply:
This is the key part -- the lock guard is passed to the function call, ensuring the `quit` part doesn't change in the meantime.